manticore/native/cpu/abstractcpu.py

```diff
@@ -840,17 +889,25 @@ def execute(self):
                 register_logger.debug(l)
 
         try:
-            implementation = getattr(self, name, None)
-
-            if implementation is not None:
-                implementation(*insn.operands)
-
-            else:
-                text_bytes = ' '.join('%02x' % x for x in insn.bytes)
-                logger.warning("Unimplemented instruction: 0x%016x:\t%s\t%s\t%s",
-                               insn.address, text_bytes, insn.mnemonic, insn.op_str)
+            if self._concrete and 'SYSCALL' in name:
+                self.emu.sync_unicorn_to_manticore()
+            if self._concrete and 'SYSCALL' not in name:
```

Comment:
It might be good to actually change this part a bit and have just one `if self._concrete` and a `if 'SYSCALL' in name: ... else: ...` inside.

```suggestion
            elif self._concrete and 'SYSCALL' not in name:
```

Reply:
I've actually made both of the optimizations you describe here independently without realizing until I tested them that they subtly break the logic. You could refactor it to have a nested `if` at the top, but then you need to change the final `else` to `if not self.concrete or 'SYSCALL' in name` so I don't think it makes any substantial improvements for readability.